js/privatebin.js

```diff
@@ -5558,6 +5624,16 @@ jQuery.PrivateBin = (function($, RawDeflate) {
             try {
                 Model.getPasteId();
             } catch (e) {
+
+                // check if protected pathname
+                const url = new URL(window.location);
+
+                // if protected request password
+                if(url.hash.startsWith('#protected/')) {
+                    return Prompt.requestPassword();
+
+                }
```

Comment:
Rather catch (or create) a specific `Error`  for that type and use it to prompt for a password. So password promting is not done for any other errors…

Also this can again use the common logic then… (or rather this whole check would not be needed anymore.)

`Error`  could be named `DecryptionError` or `PasteRetrievalError` .

Reply:
not sure what you're suggesting. And not sure why we placed this exactly here. However this is much rather a super leightweight routing implementation, so if anything may be changed, that may be placed way earlier in the code itself. Again not  sure why we placed it exactly here, maybe there was some logic that was necessary to be executed before.

This code is over 2 years old.